smart-contracts/contracts/cl-vault/src/contract.rs

```diff
@@ -227,248 +220,15 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
-    let old_vault_config = OLD_VAULT_CONFIG.load(deps.storage)?;
-    let new_vault_config = VaultConfig {
-        performance_fee: old_vault_config.performance_fee,
-        treasury: old_vault_config.treasury,
-        swap_max_slippage: old_vault_config.swap_max_slippage,
-        dex_router: deps.api.addr_validate(msg.dex_router.as_str())?,
-    };
-
-    OLD_VAULT_CONFIG.remove(deps.storage);
-    VAULT_CONFIG.save(deps.storage, &new_vault_config)?;
-
-    MIGRATION_STATUS.save(deps.storage, &MigrationStatus::Open)?;
-
-    // Declare response object as mut
-    let mut response = Response::new().add_attribute("migrate", "successful");
-
-    // Conditionally add a bank send message if the strategist rewards state is not empty
-    #[allow(deprecated)]
-    let strategist_rewards = STRATEGIST_REWARDS.load(deps.storage)?;
-    if !strategist_rewards.is_empty() {
-        let bank_send_msg = BankMsg::Send {
-            to_address: new_vault_config.treasury.to_string(),
-            amount: sort_tokens(strategist_rewards.coins()),
-        };
-        response = response.add_message(bank_send_msg);
-    }
-    // Remove the state
-    #[allow(deprecated)]
-    STRATEGIST_REWARDS.remove(deps.storage);
-
-    Ok(response)
-}
-
-#[cfg(test)]
-mod tests {
-    use cosmwasm_std::{
-        coin,
-        testing::{mock_dependencies, mock_env},
-        Addr, Coin, CosmosMsg, Decimal, SubMsg, Uint128,
-    };
-    use prost::Message;
-    use std::str::FromStr;
-
-    #[allow(deprecated)]
-    use crate::{
-        rewards::CoinList, state::USER_REWARDS,
-        test_tube::initialize::initialize::MAX_SLIPPAGE_HIGH,
-    };
-    use crate::{
-        state::OldVaultConfig,
-        test_tube::initialize::initialize::{DENOM_BASE, DENOM_QUOTE, DENOM_REWARD},
+pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
```

Comment:
Looks like this will fix the issue. But should we do it this way? Or just trigger a rebalance that changes the join_time of the positon?